cli/popper/runner_slurm.py

```diff
@@ -0,0 +1,127 @@
+import os
+from subprocess import PIPE, Popen, STDOUT, SubprocessError
+
+import docker
+
+from popper import utils as pu
+from popper import scm
+from popper.cli import log as log
+from popper.runner import StepRunner as StepRunner
+from popper.runner_host import DockerRunner as HostDockerRunner
+from popper.runner_host import HostRunner
+
+
+class SlurmRunner(StepRunner):
+
+    spawned_processes = set()
+
+    def __init__(self, config):
+        super(SlurmRunner, self).__init__(config)
+
+    def __exit__(self, exc_type, exc, traceback):
+        SlurmRunner.spawned_processes = set()
+
+    def exec_srun_cmd(self, cmd):
```

Comment:
given that this is very similar to the code in `runner_host.HostRunner`, maybe we can refactor the code in `runner_host.HostRunner` so that there's a member function that can be invoked here, that has all this logic and the only thing we pass that is different is the leading `srun` to the command.

Reply:
Can we have the function in `utils.py` like it was before ?

Reply:
sure, that'd work. I thought of having it in the parent class because currently there's no other code that makes us of it. But having in utils.py is fine